src/components/MessagesList/MessagesGroup/Message/MessagePart/ReactionsList.vue

```diff
@@ -0,0 +1,191 @@
+			</h2>
+			<template v-if="Object.keys(detailedReactions).length > 0">
+				<div class="reactions-list__navigation">
+					<NcButton :class="{'active' : reactionFilter === ''}"
+						type="tertiary"
+						@click="handleTabClick('')">
+						{{ t('spreed', 'All ({count})', { count: flatReactions.length }) }}
+					</NcButton>
+					<NcButton v-for="reaction in reactionsSorted"
+						:key="reaction"
+						:class="{'active' : reactionFilter === reaction}"
+						type="tertiary"
+						@click="handleTabClick(reaction)">
+						{{ reaction }} {{ detailedReactions[reaction].length }}
+					</NcButton>
+				</div>
+
+				<NcListItemIcon v-for="item in filteredReactions ?? flatReactions"
+					:key="item.actorDisplayName"
+					:name="getDisplayNameForReaction(item)">
+					<span>
+						{{ reactionFilter === '' ? item.reaction.join('') : reactionFilter }}
+					</span>
+				</NcListItemIcon>
+			</template>
+			<NcLoadingIcon v-else :size="64" />
+		</div>
+	</NcModal>
+</template>
+
+<script>
+import NcButton from '@nextcloud/vue/dist/Components/NcButton.js'
+import NcListItemIcon from '@nextcloud/vue/dist/Components/NcListItemIcon.js'
+import NcLoadingIcon from '@nextcloud/vue/dist/Components/NcLoadingIcon.js'
+import NcModal from '@nextcloud/vue/dist/Components/NcModal.js'
+
+import { ATTENDEE } from '../../../../../constants.js'
+import { useGuestNameStore } from '../../../../../stores/guestName.js'
+
+export default {
+
+	name: 'ReactionsList',
+
+	components: {
+		NcModal,
+		NcLoadingIcon,
+		NcListItemIcon,
+		NcButton,
+	},
+
+	props: {
+		token: {
+			type: String,
+			required: true,
+		},
+
+		detailedReactions: {
+			type: Object,
+			default: () => {},
+		},
+
+		reactionsSorted: {
+			type: Array,
+			default: () => [],
+		},
+	},
+
+	emits: ['close'],
+
+	setup() {
+		return {
+			guestNameStore: useGuestNameStore(),
+		}
+	},
+
+	data() {
+		return {
+			reactionFilter: '',
+		}
+	},
+
+	computed: {
+		container() {
+			return this.$store.getters.getMainContainerSelector()
+		},
+
+		flatReactions() {
+			const mergedReactionsMap = {}
+
+			Object.entries(this.detailedReactions).forEach(([reaction, actors]) => {
+				actors.forEach(actor => {
+					const key = `${actor.actorId}-${actor.actorType}`
+					if (mergedReactionsMap[key]) {
+						mergedReactionsMap[key].reaction.push(reaction)
+					} else {
+						mergedReactionsMap[key] = {
+							actorDisplayName: actor.actorDisplayName,
+							actorId: actor.actorId,
+							actorType: actor.actorType,
+							reaction: [reaction]
+						}
+					}
+				})
+			})
+
+			return Object.values(mergedReactionsMap)
+		},
+
+		filteredReactions() {
+			if (!this.reactionFilter) {
+				return this.flatReactions
+			} else {
+				return this.flatReactions.filter(reaction => reaction.reaction.includes(this.reactionFilter))
+			}
+		},
+
+	},
+
+	methods: {
+		closeModal() {
+			this.$emit('close')
+		},
+
+		getDisplayNameForReaction(reactingParticipant) {
```

Comment:
Also would be better to include it in flatReactions, to call only once / on list update, instead of every tab change